src/displayapp/screens/ScoreApp.cpp

```diff
@@ -0,0 +1,80 @@
+#include "ScoreApp.h"
+#include "Screen.h"
+#include "Symbols.h"
+#include "lvgl/lvgl.h"
+
+using namespace Pinetime::Applications::Screens;
+
+
+static void btnEventHandler(lv_obj_t* obj, lv_event_t event) {
+  ScoreApp* screen = static_cast<ScoreApp*>(obj->user_data);
+  screen->OnButtonEvent(obj, event);
+}
+
+ScoreApp::widget_t ScoreApp::_createButton(lv_align_t alignment, uint8_t x, uint8_t y, uint8_t width, uint8_t height, const char text[]){
+
+  widget_t button;
+
+  button.button = lv_btn_create(lv_scr_act(), nullptr);
+  button.button->user_data = this;
+  lv_obj_set_event_cb(button.button, btnEventHandler);
+  lv_obj_align(button.button, lv_scr_act(), alignment, x, y);
+  lv_obj_set_width(button.button, width);
+  lv_obj_set_height(button.button, height);
+  button.label = lv_label_create(button.button, nullptr);
+  lv_label_set_text(button.label, text);
+
+  return button;
+}
+
+ScoreApp::ScoreApp(DisplayApp* app) : Screen(app) {
+    
+  btnMyScore = _createButton(LV_ALIGN_IN_TOP_LEFT, 0, 0, scoreWidth, scoreHeight, "0");
+  lv_obj_set_style_local_text_font(btnMyScore.label, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, &jetbrains_mono_76);
+  lv_obj_set_style_local_text_color(btnMyScore.label, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_GRAY);
+
+  btnYourScore = _createButton(LV_ALIGN_IN_TOP_LEFT, 0, +scoreHeight, scoreWidth, scoreHeight, "0");
+  lv_obj_set_style_local_text_font(btnYourScore.label, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, &jetbrains_mono_76);
+  lv_obj_set_style_local_text_color(btnYourScore.label, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_GRAY);
+
+  lv_label_set_text_fmt(btnMyScore.label, "%d", myScore);
+  lv_label_set_text_fmt(btnYourScore.label, "%d", yourScore);
+  
+  btnMyScoreMinus = _createButton(LV_ALIGN_IN_TOP_RIGHT, 0, 0, minusWidth, minusHeight, "-1");
+  btnYourScoreMinus = _createButton(LV_ALIGN_IN_TOP_RIGHT, 0, displayHeight-minusHeight, minusWidth, minusHeight, "-1");
+  
+  btnReset = _createButton(LV_ALIGN_IN_TOP_RIGHT, 0, displayHeight/2, resetWidth, resetHeight, "Rst");
+
+  taskRefresh = lv_task_create(RefreshTaskCallback, LV_DISP_DEF_REFR_PERIOD, LV_TASK_PRIO_MID, this);
+}
+
+ScoreApp::~ScoreApp() {
+  lv_task_del(taskRefresh);
+  lv_obj_clean(lv_scr_act());
+}
+
+void ScoreApp::Refresh() {
+  lv_label_set_text_fmt(btnMyScore.label, "%d", myScore);
+  lv_label_set_text_fmt(btnYourScore.label, "%d", yourScore);
+}
+
```

Comment:
This app doesn't require `Refresh()` at all at least currently. You can remove the task and update the labels in `OnButtonEvent()` instead

Reply:
Yes, I did not understood that refresh is used mainly for synchronous/external event and asynchronous event like buttons and touch can be handled directly.
Corrected in https://github.com/toitoinou/Pinetime/commit/b5a762a1de2682963eb98bda42a06ee710741935